extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketches.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.datasketches.quantiles;
+
+import org.apache.druid.java.util.common.ISE;
+
+public final class DoublesSketches
+{
+  /**
+   * Runs the given task that updates a Doubles sketch backed by a direct byte buffer. This method intentionally
+   * accepts the update task as a {@link Runnable} instead of accpeting parameters of a sketch and other values
+   * needed for the update. This is to avoid any potential performance impact especially when the sketch is updated
```

Comment:
did you measure if this approach has any perf impact?

Reply:
I ran the benchmark I added in https://github.com/apache/druid/pull/11574/commits/7be41051242f16db225dfce48f86adc9b1966c3a. It doesn't seem to have any difference.

```
// 20: doubles sketches
query: "SELECT APPROX_QUANTILE_DS(sumFloatNormal, 0.5), DS_QUANTILES_SKETCH(maxLongUniform) FROM foo"

master

Benchmark              (query)  (rowsPerSegment)  (vectorize)  Mode  Cnt    Score   Error  Units
SqlBenchmark.querySql       20           5000000        false  avgt   15  341.659 ± 0.932  ms/op
SqlBenchmark.querySql       20           5000000        force  avgt   15  446.572 ± 3.279  ms/op

PR

Benchmark              (query)  (rowsPerSegment)  (vectorize)  Mode  Cnt    Score   Error  Units
SqlBenchmark.querySql       20           5000000        false  avgt   15  341.956 ± 2.478  ms/op
SqlBenchmark.querySql       20           5000000        force  avgt   15  442.048 ± 1.420  ms/op
```

Interestingly, the query was _slower_ when the vectorization was on. AFAIT, the difference is that the non-vectorized version uses `HeapUpdateDoublesSketch` while the vectorized one uses `DirectUpdateDoublesSketch`. Attaching flame graphs.

Non-vectorized
![Screenshot from 2021-08-24 21-02-35](https://user-images.githubusercontent.com/2322288/130724807-18b430fb-ca4c-4644-aae3-20612b0cd0cf.png)

Vectorized
![Screenshot from 2021-08-24 21-02-57](https://user-images.githubusercontent.com/2322288/130724816-74bdabe7-b7cf-439f-855a-2dd7dbd6460e.png)

Reply:
I tested a couple of more queries that produce more than one row. Looking at these results, the change in this PR doesn't seem to introduce huge overhead. However, the performance of these queries does seem interesting when vectorization is on. For query 2, it used `DirectUpdateDoublesSketch` for both when vectorization was on and off, so the result seems reasonable as the query was faster with vectorization. However, for query 3, even though both used `DirectUpdateDoublesSketch`, the query was slower with vectorization. This seems because `MemoryOpenHashTable.copyTo()` was more expensive than `ByteBufferHashTable.adjustTableWhenFull()`. I'm not 100% sure about this yet, so will dig further. 

```
query 2: 
SELECT dimZipf, APPROX_QUANTILE_DS(sumFloatNormal, 0.5), DS_QUANTILES_SKETCH(maxLongUniform)
FROM foo
GROUP BY 1

master

Benchmark              (query)  (rowsPerSegment)  (vectorize)  Mode  Cnt    Score   Error  Units
SqlBenchmark.querySql       20           5000000        false  avgt   15  617.615 ± 4.681  ms/op
SqlBenchmark.querySql       20           5000000        force  avgt   15  512.804 ± 3.785  ms/op

PR

Benchmark              (query)  (rowsPerSegment)  (vectorize)  Mode  Cnt    Score   Error  Units
SqlBenchmark.querySql       20           5000000        false  avgt   15  625.830 ± 8.074  ms/op
SqlBenchmark.querySql       20           5000000        force  avgt   15  533.246 ± 3.985  ms/op
```

```
query 3: 
SELECT dimSequential, dimZipf, APPROX_QUANTILE_DS(sumFloatNormal, 0.5)
FROM foo
GROUP BY 1, 2

* To avoid ResourceLimitExceededException, the buffer size and rowsPerSegment were set to 2GB and 100000, respectively. 

master

Benchmark              (query)  (rowsPerSegment)  (vectorize)  Mode  Cnt    Score   Error  Units
SqlBenchmark.querySql       20            100000        false  avgt   15  391.651 ± 1.849  ms/op
SqlBenchmark.querySql       20            100000        force  avgt   15  671.631 ± 1.640  ms/op

PR

Benchmark              (query)  (rowsPerSegment)  (vectorize)  Mode  Cnt    Score   Error  Units
SqlBenchmark.querySql       20            100000        false  avgt   15  395.111 ± 1.488  ms/op
SqlBenchmark.querySql       20            100000        force  avgt   15  672.921 ± 2.680  ms/op
```